File: Java/Chapter12/TvFactory.java
public class TvFactory {

    static ITv createTv(String tvSpec) 
    	throws TvException {

	if (tvSpec.equals("プラズマ"))
	    return new PlasmaTv();
	else if (tvSpec.equals("液晶"))
	    return new LcdTv();
	else
	    throw new TvException(
		tvSpec + " は利用不可．");
    }
}

File: Python/Chapter05/tv_factory.py
from plasma_tv import PlasmaTv
from lcd_tv import LcdTv
from crt_tv import CrtTv
from tv_exception import TvException

class TvFactory:
    
    @classmethod
    def createtv(cls, tvspec):
        if tvspec == 'プラズマ':
            return PlasmaTv()
        elif tvspec == '液晶':
            return LcdTv()
        else:
            raise TvException(tvspec + 'は利用不可．')
